src/book/04-guides/01-nubots/02-camera-calibration.mdx

```diff
@@ -0,0 +1,74 @@
+
+   Next you will need to figure out which is left and which is right. Change the serial number of one camera (this will ensure it is not found when running cameratest). Enable compressed images in `NUsight.yaml`. Run `./cameratest` and test which camera is working, simply by covering one camera and seeing the resulting image in the NUsight vision tab. Alternatively you could unplug one camera rather than change the serial number. If the serial number is on the wrong camera configuration file, switch them.
+
+5. Check the focus of the lenses using the NUsight vision tab. If it is not sharp, focus the camera.
+
+6. Check the transform matrix from the left camera to the pitch motor (Hpc) is correct in the `Left.yaml` file.
+
+7. Run `./cameratest` on the robot and hold up a checkerboard, moving it around the area to cover most of the view over time. Make sure you cover all of the available image, and make sure that you tilt the board at different angles from the lens.
+
+   ![A white board with black circles in rows.](./images/asymmetric-circle-grid.svg 'Asymmetric circles grid used for calibration.')
+
+8. Stop the binary and copy the nbs file created into the NUbots directory on your computer by using the command `scp <robot address>:<file path> .` on your computer in your NUbots directory. The file will be in the `log/cameratest/` folder on the robot. The name will correspond to the date and time it was created, according to the robot. Check on the robot first using `ls` to determine the name. The name can be changed by the command `mv <old path> <new path>`.
+
+   The robot can be turned off at this point. The recording you just made should be on your computer in the NUbots directory on your computer. Check that it is before continuing to the next steps.
+
+9. Run the following commands if you have not done this before
+
+   ```bash
+   sudo apt install protobuf-compiler libprotobuf-dev
+   sudo -H pip3 install --upgrade pip
+   sudo -H pip3 install tensorflow opencv-contrib-python ruamel.yaml
+   ```
+
+10. Calibrate the cameras by running
+
+    ```bash
+    ./b nbs calibrate_cameras <nbs file name> -c <camera config folder>
+    ```
+
+    `<nbs file name>` is the name of the nbs file you just created. `<camera config folder>` is the path to the folder with the two camera config files `Left.yaml` and `Right.yaml`. These exist in `./module/input/Camera/data/config/<robot name>/Cameras/`.
+
+Once the calibration is done, the values in the camera configuration files should be changed. Commit these changes in a new branch on GitHub to ensure they are not lost. If you are calibrating cameras on other robots, repeat the process with them. Once all the configuration that is wanted to be done is complete, make sure it is all commited and pushed on a branch and then make a pull request to get the changes into the master branch.
```

Comment:
You also should validate the configuration on a robot properly. Have the robot standing and make sure the blue line follows a constant height on the wall down the image. You can use the bricks for this, stand the robot near a brick wall and make sure the blue line stays in the same place along the bricks. Also when the robot is looking at the wall, the axis it's looking down the distance between the points on the wall should be the same distance between the cameras

Reply:
" Also when the robot is looking at the wall, the axis it's looking down the distance between the points on the wall should be the same distance between the cameras"
These are words... But I can't figure out a meaning.

Reply:
@TrentHouliston any clarification? D:

Reply:
@TrentHouliston

Reply:
Right, sorry. What I mean by this is that when you have the two blue lines on the wall, if the robot is directly facing the wall then the same point on the blue line on each should be the distance between the eyes. So it might be easier to do this with the compass, Face the robot at the wall, and somehow get the compass to face forward where the horizon and compass intersect should be the eye distance apart

Reply:
That's still a little tricky. Maybe just worry about the horizon following a line on the brick wall. The other one is important because you're validating the yaw is correct but it's hard to explain what I mean.